batcher/aligned-batcher/src/lib.rs

```diff
@@ -781,9 +783,20 @@ impl Batcher {
         zk_utils::is_verifier_disabled(*disabled_verifiers, verifier)
     }
 
-    // Checks user has sufficient balance for paying all its the proofs in the current batch.
-    fn check_min_balance(&self, user_proofs_in_batch: usize, user_balance: U256) -> bool {
-        let min_balance = U256::from(user_proofs_in_batch) * U256::from(MIN_FEE_PER_PROOF);
+    // Checks user has sufficient balance for paying all the users proofs in the current batch.
+    fn check_min_balance(
+        &self,
+        user_min_fee: U256,
+        user_proofs_in_batch: usize,
+        user_balance: U256,
+        user_max_fee: U256,
+    ) -> bool {
+        // `user_min_fee` is the minimum `max_fee` the user submitted to the batcher and represents the maximium price that the user will pay for each submitted proof. We define 'user_min_fee' as an upper bound for the proof submission cost of the user, and use it to validate the user's balance has enough fund available to pay for all submitted proofs.
+        let mut min_fee = user_min_fee;
+        if user_min_fee == U256::max_value() {
+            min_fee = user_max_fee
+        }
+        let min_balance: U256 = U256::from(user_proofs_in_batch) * min_fee;
```

Comment:
Maybe I am wrong here but why don't we move the `min_balance` check after the `msg_max_fee` check? This way, in the `min_balance` check we'll verify against the `msg_max_fee` directly (as it will become the new min_fee if everything goes ok), we'll save ourselves this if:

```rust
        if user_min_fee == U256::max_value() {
            min_fee = user_max_fee
        }
```
It is hard to understand that `U256::max_value()` is the default value.